src/org/parosproxy/paros/network/SSLConnector.java

```diff
@@ -529,17 +534,28 @@ public Socket createSocket(Socket socket, String host, int port,
 	 * @throws IOException
 	 */
 	public Socket createTunnelServerSocket(String targethost, Socket socket) throws IOException {
+		InetAddress listeningAddress = socket.getLocalAddress();
 		// ZAP: added host name parameter
-		SSLSocket s = (SSLSocket) getTunnelSSLSocketFactory(targethost).createSocket(socket, socket
+		SSLSocket s = (SSLSocket) getTunnelSSLSocketFactory(targethost, listeningAddress).createSocket(socket, socket
 				.getInetAddress().getHostAddress(), socket.getPort(), true);
 		
 		s.setUseClientMode(false);
 		s.startHandshake();
 		return s;
 	}
 
-	// ZAP: added new ServerSocketFaktory with support of dynamic SSL certificates
+
+	/**
+	 * @deprecated (TODO add version) No longer used/needed.
+	 */
+	@Deprecated
+	// ZAP: added new ServerSocketFactory with support of dynamic SSL certificates

```

Comment:
Add the ZAP comment into the javadoc so that the annotation appears right before the applicable code?

Reply:
I added the // ZAP comment into the javadoc